.tekton/tasks/changed-dirs.yaml

```diff
@@ -0,0 +1,47 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: changed-dirs
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/displayName: "Lists changed files"
+    tekton.dev/platforms: "linux/amd64"
+spec:
+  description: Produces a list of changed top-level directories
+  params:
+    - name: pr_number
+      type: string
+    - name: utils_image
+      type: string
+  results:
+    - name: changed_directories
+      type: array
+  steps:
+    - name: list-changed-files
+      image: $(params.utils_image)
+      env:
+        - name: GITHUB_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: "{{ git_auth_secret }}"
+              key: "git-provider-token"
+      script: |
+        #!/bin/bash
+        set -o errexit
+        set -o nounset
+        set -o pipefail
+
+        dirs=()
+        # collect all directories changed in a PR
+        for path in $(gh pr view "https://github.com/konflux-ci/build-definitions/pull/$(params.pr_number)" --json files --jq '.files.[].path'); do
+          dirs+=("${path%%/*}")
+        done
+        # deduplicate dirs
+        readarray -t dirs < <(printf '%s\n' "${dirs[@]}" | sort -u)
+        # join $dirs with '","', e.g. (a b c) -> 'a","b","c","'
+        escaped_json="$(printf '%s\",\"' "${dirs[@]}")"
+        # add the square brackets and remove the trailing ',"'
+        escaped_json="[\"${escaped_json::-2}]"
+        echo -n "${escaped_json}" > "$(results.changed_directories.path)"
```

Comment:
The number of directories in this repo will blow past the result limit.

```
$ find . -type d | wc -c | numfmt --to=iec
11K
```

Reply:
It's only creating an array of top-level directories, so the current size is:

```
$ ls -d */|jq -cRn '[inputs]'|wc -c
107
```

Reply:
Great!